### PR TITLE
Update io.pyi

### PR DESCRIPTION
**Pull request by baterflyrity**
See corresponding [issue](https://github.com/python/typeshed/issues/4906#issue-779374573).

### CODE REVIEW

**Comment by Akuli** [comment]
Maybe this is related: https://github.com/python/mypy/issues/5876

File: stdlib/3/io.pyi [hunk 1]
```diff
@@ -34,12 +34,14 @@ class IOBase:
     def flush(self) -> None: ...
     def isatty(self) -> bool: ...
     def readable(self) -> bool: ...
+    def read(self, *args: Any, **kwargs: Any) -> Any: ...
```
Comment (Akuli) [inline stdlib/3/io.pyi:37]:
```suggestion
    read: Callable[..., Any]
    def read(self, *args: Any, **kwargs: Any) -> Any: ...
```

File: stdlib/3/io.pyi [hunk 2]
```diff
@@ -34,12 +34,14 @@ class IOBase:
...
     def readlines(self, __hint: int = ...) -> List[bytes]: ...
     def seek(self, __offset: int, __whence: int = ...) -> int: ...
     def seekable(self) -> bool: ...
     def tell(self) -> int: ...
     def truncate(self, __size: Optional[int] = ...) -> int: ...
     def writable(self) -> bool: ...
+    def write(self, *args: Any, **kwargs: Any) -> Any: ...
```
Comment (Akuli) [inline stdlib/3/io.pyi:44]:
```suggestion
    write: Callable[..., Any]
```

**Review by Akuli — CHANGES_REQUESTED** [review]
This should fix the mypy_primer errors. I tried doing what werkzeug does, but I didn't check if other errors get fixed with this.

**Comment by baterflyrity** [comment]
@Akuli , thanks for help. Useful link and commit, hope it will help.

ps
Also one can try for similar issue next links:  [partially arbitrary parameters](https://github.com/python/mypy/issues/5876#issuecomment-706396563) and [property hack](https://github.com/python/mypy/issues/9560#issuecomment-705955103).

**Comment by Akuli** [comment]
Don't know if mypy_primer is showing outdated results above

**Comment by baterflyrity** [comment]
@Akuli , so what can you suggest?

**Comment by Akuli** [comment]
I would try making an empty commit to rerun mypy_primer. Don't know if there's a better way.

**Comment by baterflyrity** [comment]
@Akuli , very nice, according to [mypy_primer](https://github.com/python/typeshed/pull/4912/checks?check_run_id=1661452668#step:3:324) no diffs found. So what is needed next to merge this PR and close #4906 ?

**Comment by Akuli** [comment]
A typeshed collaborator (not me, I'm just a contributor) has to review this, and if it looks good to the collaborator, they will likely merge it and close the issue.

**Review by Akuli — APPROVED** [review]

**Review by srittau — APPROVED** [review]
The change seems correct to me and the best solution. I am still a bit wary that this might cause unforeseen problems, but the mypy_primer output makes me optimistic. Let's try this an be prepared to revert if it causes problems.

**Pull request merged**